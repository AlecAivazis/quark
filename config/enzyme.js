// external imports
import Enzyme from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import browserEnv from 'browser-env'

// add a request animation frame polyfill
import raf from 'raf'
raf.polyfill()

// enable the react 16 adapter for enzyme
Enzyme.configure({ adapter: new Adapter() })

// add various browser mocks
browserEnv()
