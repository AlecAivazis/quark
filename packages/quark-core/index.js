export * from './components';
import * as _styles from './styles';
export { _styles as styles };