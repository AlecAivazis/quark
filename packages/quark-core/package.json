{
    "name": "quark-core",
    "version": "1.0.6",
    "module": "index.js",
    "main": "index.js",
    "dependencies": {
        "react-native-web": "0.3.4",
        "svgs": "^3.1.1"
    },
    "peerDependencies": {
        "react": "^16.3.0"
    }
}
