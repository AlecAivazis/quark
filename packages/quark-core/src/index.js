// @flow
export * from './components'
export * as styles from './styles'
