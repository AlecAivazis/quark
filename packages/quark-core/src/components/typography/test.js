// external imports
import React from 'react'
import { Text } from 'react-native'
import { mount } from 'enzyme'
// local imports
import { H1, H2, H3, Title, Subtitle, Link, Monospace } from 'quark-core'

it('H1 renders', () => expect(mount(<H1>hello</H1>)).toMatchSnapshot())

it('H2 renders', () => expect(mount(<H2>hello</H2>)).toMatchSnapshot())

it('H3 renders', () => expect(mount(<H3>hello</H3>)).toMatchSnapshot())

it('Title renders', () => expect(mount(<Title>hello</Title>)).toMatchSnapshot())

it('Subtitle renders', () => expect(mount(<Subtitle>hello</Subtitle>)).toMatchSnapshot())

it('Link renders', () => expect(mount(<Link>hello</Link>)).toMatchSnapshot())

it('Monospace renders', () => expect(mount(<Monospace>hello</Monospace>)).toMatchSnapshot())
