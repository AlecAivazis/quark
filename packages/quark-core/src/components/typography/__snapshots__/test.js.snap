// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`H1 renders 1`] = `
<H1>
  <InnerText
    style={
      Object {
        "marginBottom": NaN,
      }
    }
  >
    <GetTheme>
      <Text
        style={
          Object {
            "color": "#48484D",
            "margin": 0,
            "marginBottom": NaN,
          }
        }
      >
        <div
          className="rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-1471scf rn-font-1lw9tu2 rn-fontFamily-10u92zi rn-fontSize-1b43r93 rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginLeft-11wrixw rn-marginBottom-1kk9pjb rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-textDecoration-bauka4 rn-whiteSpace-q42fyq rn-wordWrap-qvutc0"
          dir="auto"
          style={
            Object {
              "color": "rgba(72,72,77,1)",
            }
          }
        >
          hello
        </div>
      </Text>
    </GetTheme>
  </InnerText>
</H1>
`;

exports[`H2 renders 1`] = `
<H2>
  <InnerText
    style={
      Object {
        "marginBottom": NaN,
      }
    }
  >
    <GetTheme>
      <Text
        style={
          Object {
            "color": "#48484D",
            "margin": 0,
            "marginBottom": NaN,
          }
        }
      >
        <div
          className="rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-1471scf rn-font-1lw9tu2 rn-fontFamily-10u92zi rn-fontSize-1b43r93 rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginLeft-11wrixw rn-marginBottom-1kk9pjb rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-textDecoration-bauka4 rn-whiteSpace-q42fyq rn-wordWrap-qvutc0"
          dir="auto"
          style={
            Object {
              "color": "rgba(72,72,77,1)",
            }
          }
        >
          hello
        </div>
      </Text>
    </GetTheme>
  </InnerText>
</H2>
`;

exports[`H3 renders 1`] = `
<H3>
  <InnerText
    style={
      Object {
        "marginBottom": NaN,
      }
    }
  >
    <GetTheme>
      <Text
        style={
          Object {
            "color": "#48484D",
            "margin": 0,
            "marginBottom": NaN,
          }
        }
      >
        <div
          className="rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-1471scf rn-font-1lw9tu2 rn-fontFamily-10u92zi rn-fontSize-1b43r93 rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginLeft-11wrixw rn-marginBottom-1kk9pjb rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-textDecoration-bauka4 rn-whiteSpace-q42fyq rn-wordWrap-qvutc0"
          dir="auto"
          style={
            Object {
              "color": "rgba(72,72,77,1)",
            }
          }
        >
          hello
        </div>
      </Text>
    </GetTheme>
  </InnerText>
</H3>
`;

exports[`Link renders 1`] = `
<Link>
  <InnerText
    style={
      Array [
        undefined,
        84,
        undefined,
      ]
    }
  >
    <GetTheme>
      <Text
        style={
          Object {
            "color": "#48484D",
            "margin": 0,
          }
        }
      >
        <div
          className="rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-1471scf rn-font-1lw9tu2 rn-fontFamily-10u92zi rn-fontSize-1b43r93 rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-textDecoration-bauka4 rn-whiteSpace-q42fyq rn-wordWrap-qvutc0"
          dir="auto"
          style={
            Object {
              "color": "rgba(72,72,77,1)",
            }
          }
        >
          hello
        </div>
      </Text>
    </GetTheme>
  </InnerText>
</Link>
`;

exports[`Monospace renders 1`] = `
<Monospace>
  <InnerText
    style={
      Array [
        undefined,
        84,
        undefined,
      ]
    }
  >
    <GetTheme>
      <Text
        style={
          Object {
            "color": "#48484D",
            "margin": 0,
          }
        }
      >
        <div
          className="rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-1471scf rn-font-1lw9tu2 rn-fontFamily-10u92zi rn-fontSize-1b43r93 rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-textDecoration-bauka4 rn-whiteSpace-q42fyq rn-wordWrap-qvutc0"
          dir="auto"
          style={
            Object {
              "color": "rgba(72,72,77,1)",
            }
          }
        >
          hello
        </div>
      </Text>
    </GetTheme>
  </InnerText>
</Monospace>
`;

exports[`Subtitle renders 1`] = `
<Subtitle>
  <InnerText
    style={
      Array [
        undefined,
        82,
        undefined,
      ]
    }
  >
    <GetTheme>
      <Text
        style={
          Object {
            "color": "#48484D",
            "margin": 0,
          }
        }
      >
        <div
          className="rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-1471scf rn-font-1lw9tu2 rn-fontFamily-10u92zi rn-fontSize-1b43r93 rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-textDecoration-bauka4 rn-whiteSpace-q42fyq rn-wordWrap-qvutc0"
          dir="auto"
          style={
            Object {
              "color": "rgba(72,72,77,1)",
            }
          }
        >
          hello
        </div>
      </Text>
    </GetTheme>
  </InnerText>
</Subtitle>
`;

exports[`Title renders 1`] = `
<Title>
  <InnerText
    style={
      Array [
        undefined,
        81,
        undefined,
      ]
    }
  >
    <GetTheme>
      <Text
        style={
          Object {
            "color": "#48484D",
            "margin": 0,
          }
        }
      >
        <div
          className="rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-1471scf rn-font-1lw9tu2 rn-fontFamily-10u92zi rn-fontSize-1b43r93 rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-textDecoration-bauka4 rn-whiteSpace-q42fyq rn-wordWrap-qvutc0"
          dir="auto"
          style={
            Object {
              "color": "rgba(72,72,77,1)",
            }
          }
        >
          hello
        </div>
      </Text>
    </GetTheme>
  </InnerText>
</Title>
`;
